content/IA-update_connections/connection-state-and-recovery.textile

```diff
@@ -0,0 +1,292 @@
+
+blang[javascript].
+  In the browser environment, if a callback is provided in the @recover@ option, when the @window.beforeunload@ event fires, the connection details, including the "@recoveryKey@":/api/realtime-sdk/connection#recovery-key, are stored in the "browser's sessionStorage":https://www.w3.org/TR/webstorage/. The provided @recover@ callback is then invoked whenever the connection state can be recovered and just before a connection is established, passing in the "@LastConnectionDetails@":/api/realtime-sdk/connection#last-connection-details. The callback is then responsible for confirming whether the connection state should be recovered or not. For example, it is common to recover connection state when the page is reloaded but not for different pages the user has navigated to. The callback allows the developer to decide if the connection should be recovered or not at the time the new connection is established by inspecting the "@LastConnectionDetails@":/api/realtime-sdk/connection#last-connection-details and evaluating that against any other application state. Below are two examples:
+
+  * **Always recover** - always recover the previous connection state if possible
+
+  ```[jsall]
+    const ably = new Ably.Realtime({
+      authUrl: '/obtainToken',
+      recover: (_, cb) => { cb(true); }
+    });
+  ```
+
+  * **Sometimes recover** - recover the previous connection state conditionally based on some logic
+
+  ```[jsall]
+    const ably = new Ably.Realtime({
+      authUrl: '/obtainToken',
+      recover: (lastConnectionDetails, cb) => {
+        /* Only recover if the current path hasn't changed, start a
+         * fresh connection if it has. This is just an example, you
+         * can use whatever logic your app requires */
+        if (lastConnectionDetails.location.href === document.location.href) {
+          cb(true); /* recover connection */
+        } else {
+          cb(false); /* do not recover connection */
+        }
+      }
+    });
+  ```
+
+  Please note that as "@sessionStorage@":https://www.w3.org/TR/webstorage/ is used to persist the @LastConnectionDetails@ between page reloads, it is only available for pages in the same origin and top-level browsing context.
+
+  Alternatively, if it is necessary to be explicit about the connection @recoveryKey@ , the connection can be recovered by providing the last value of the connection's @recoveryKey@ value in the "client options":/api/realtime-sdk#client-options @recover@ attribute when instancing the library.
+
+h4. Connection recovery constraints
```

Comment:
Wasn't this meant to be a caveat box in the "recover" section?

Reply:
I think this works better as a subsection. There is already a caveat box at the start of the _recover_ section.